Felienne/hedy
# Use var_access where intended in all grammar definitions #1632

**Please fill out this template by replacing all content between < >**

< Use var_access where intended in all grammar definitions >

**Description**

< 
The grammar definitions contain 2 identical rules var and var_access which are meant to be used for variable definition and variable access, respectively. In many cases, however, the grammar definitions were not using the correct rule, e.g. print is using var instead of var_access.

The changes in this PR ensure that var and var_access are used as intended in all grammar definitions. As a consequence, the check for undefined variables is moved from the Transpiler to the TypeValidator.

 > Use present tense without a subject to describe your PR, for example: "Adds translations of levels 1 to 12 to Polish"

**Fixes < TODO in the grammars #1632 >**

< Always link the number of the issue or of the discussion that your pr concerns. >
Notable exception: adding content may be done without an accompanying issue 

**How to test**
< ​
- Validate that the grammar files use var_access and var as intended.
- Because the TypeValidator now has the ability to check for undefined variables, many errors in level defaults and tests were revealed. For example, missing single quotes in if-conditions above level 12 or references to undefined vars which suspiciously look like a typo. It seems that the examples did not comply to the intention of the grammars but worked by accident. Please examine the changed code in levels and check that the changes are valid.
>

**Checklist**
Done? Check if you have it all in place using this list*
 ​

* If you're unsure about any of these, don't hesitate to ask. We're here to help!
